src/components/SignOut/SignOut.js

```diff
@@ -1,14 +1,17 @@
 import React from "react";
-import { Link } from "react-router-dom";
+import { useHistory } from "react-router-dom";
 import "./SignOut.scss";
 import { auth } from "../../firebase";
 
 export default function SignOut(props) {
+  let history = useHistory();
+
   let signOutUser = async () => {
     auth
       .signOut()
       .then(() => {
-        //navigate to Login
+        props.setSignedIn(false);
+        history.push("/login");
       })
```

Comment:
I'd handle this at the ``App.js`` level with the ``onAuthStateChanged`` listener.